core/domain/exp_jobs_one_off.py

```diff
@@ -535,3 +536,41 @@ def reduce(key, values):
         # Combine all values from multiple lists into a single list
         # for that error type.
         yield (key, list(set().union(*final_values)))
+
+
+class ImageDataMigrationJob(jobs.BaseMapReduceOneOffJobManager):
+    """ One-off job for migrating the images in the exploration
+        from the GAE to GCS
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [file_models.FileSnapshotContentModel]
+
+    @staticmethod
+    def map(file_snapshot_content_model):
+        instance_id = file_snapshot_content_model.get_unversioned_instance_id()
+        filetype = instance_id[instance_id.rfind(".") + 1:]
+
+        if filetype in ALLOWED_IMAGE_EXTENSIONS:
+            filename = instance_id[instance_id.rfind("/") + 1:]
+            exploration_id = (instance_id[1:])[:(instance_id[1:]).find("/")]
+            filepath = (instance_id[1:])[(instance_id[1:]).find("/") + 1:]
+            file_model = file_models.FileModel.get_model(
+                exploration_id, filepath, False)
+
+            if file_model and file_model.deleted:
+                # content = file_model.content
```

Comment:
This is to be uncommented, so that it actually writes to GCS.

Reply:
Don't leave things commented out like this -- you'll still have to figure out the issue of how to handle stuff in localhost anyway, so this needs to be solved properly before submit.

I suggest that you follow the way Tony did it in core/controllers/editor.py (search for fs_domain.GcsFileSystem) so that it works correctly in dev and prod and so that you can actually test it (both with automated backend tests and manual testing).

Speaking of tests, this PR needs backend tests to be written before it can be merged.